lib/matplotlib/text.py

```diff
@@ -1938,101 +1965,155 @@ def __init__(self, s, xy,
+
+            If a `~matplotlib.artist.Artist` object is passed in the units are
+            fraction if it's bounding box.
+
+            If a `~matplotlib.transforms.Transform` object is passed
+            in use that to transform ``xy`` to screen coordinates
+
+            If a callable it must take a
+            `~matplotlib.backend_bases.RendererBase` object as input
+            and return a `~matplotlib.transforms.Transform` or
+            `~matplotlib.transforms.Bbox` object
+
+            If a `tuple` must be length 2 tuple of str, `Artist`,
+            `Transform` or callable objects.  The first transform is
+            used for the *x* coordinate and the second for *y*.
+
+            See :ref:`plotting-guide-annotation` for more details.
+
+            Defaults to ``'data'``
+
+        textcoords : str, `Artist`, `Transform`, callable or tuple, optional
+            The coordinate system that ``xytext`` is given, which
+            maybe different than the coordinate system used for
+            ``xy``.
+
+            All ``xycoords`` values are valid as well as the following
+            strings:
+
+            =================   =========================================
+            Property            Description
+            =================   =========================================
+            'offset points'     offset (in points) from the *xy* value
+            'offset pixels'     offset (in pixels) from the *xy* value
+            =================   =========================================
+
+            defaults to the input of ``xycoords``
+
+        arrowprops : dict, optional
+            If not None, properties used to draw a
+            `~matplotlib.patches.FancyArrowPatch` arrow between ``xy`` and
+            ``xytext``.
+
+            If `arrowprops` does not contain the key ``'arrowstyle'`` the
+            allowed keys are:
+
+            ==========   ======================================================
+            Key          Description
+            ==========   ======================================================
+            width        the width of the arrow in points
+            headwidth    the width of the base of the arrow head in points
+            headlength   the length of the arrow head in points
+            shrink       fraction of total length to 'shrink' from both ends
+            ?            any key to :class:`matplotlib.patches.FancyArrowPatch`
+            ==========   ======================================================
+
+            If the `arrowprops` contains the key ``'arrowstyle'`` the
+            above keys are forbidden.  The allowed values of
+            ``'arrowstyle'`` are:
+
+            ============   =============================================
+            Name           Attrs
+            ============   =============================================
+            ``'-'``        None
+            ``'->'``       head_length=0.4,head_width=0.2
```

Comment:
For 'historical reasons'.  Probably should clean that up eventually, but not in this PR.